src/communication/http/httpClient.ts

```diff
@@ -192,42 +257,80 @@ export abstract class HttpClient {
     const message = readProperty(err, 'response.data.message', undefined) as string | undefined;
     switch (err.response?.status) {
       case HttpStatus.BAD_REQUEST:
-        if (body !== undefined) {
-          body = JSON.stringify(body);
-          if (!this.disableDebugLogs) {
-            this.logger.debug(`invalid request sent to ${this.targetService} at ${url}. body: ${body as string}. error: ${err.message}`);
-          }
-        } else if (!this.disableDebugLogs) {
-          this.logger.debug(`invalid request sent to ${this.targetService} at ${url}. error: ${err.message}`);
+        if (!this.disableDebugLogs) {
+          this.logger.debug(
+            {
+              err,
+              url,
+              body,
+              targetService: this.targetService,
+            },
+            `invalid request error recieved from service ${this.targetService}.`
+          );
         }
         return new BadRequestError(err, message);
       case HttpStatus.NOT_FOUND:
         if (!this.disableDebugLogs) {
-          this.logger.debug(`request url not found for service ${this.targetService}, target url: ${url}, error: ${err.message}`);
+          this.logger.debug(
+            {
+              err,
+              url,
+              body,
+              targetService: this.targetService,
+            },
+            `not found error recieved from service ${this.targetService}.`
+          );
         }
         return new NotFoundError(err, message);
       case HttpStatus.CONFLICT:
         if (!this.disableDebugLogs) {
-          this.logger.debug(`request url conflicted, for service ${this.targetService}, target url: ${url}, error: ${err.message}`);
+          this.logger.debug(
+            {
+              err,
+              url,
+              body,
+              targetService: this.targetService,
+            },
+            `conflict error recieved from service ${this.targetService}.`
+          );
         }
         return new ConflictError(err, message);
       case HttpStatus.FORBIDDEN:
         if (!this.disableDebugLogs) {
-          this.logger.debug(`forbidden request sent service ${this.targetService}, target url: ${url}, error: ${err.message}`);
+          this.logger.debug(
```

Comment:
object that include attribute msg:"the str error"

Reply:
the message already includes the status code / reason